OpenRiaServices.DomainServices.Client/Framework/Silverlight/Data/EntitySet.cs

```diff
@@ -1265,7 +1272,7 @@ protected override Entity CreateEntity()
         /// <returns>The enumerator</returns>
         public new IEnumerator<TEntity> GetEnumerator()
         {
-            return ((IList<TEntity>)List).GetEnumerator();
+            return _indexedList.AsList().Cast<TEntity>().GetEnumerator();
```

Comment:
Ok, now I understand why the list was generic and why it was a good idea.
I guess iterating the set can be done quite often so not having to add Linq overhead for enumerating the list seems lika a good idea. 

Can you please change back the the list to generic so that we don't need a cast here.
(Tip - add a public "IEnumerable<T> GetEnumerator()" to IndexedList  and you dont need to use AsList here